sphinx_gallery/gen_rst.py

```diff
@@ -639,3 +641,44 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         print("{0} ran in : {1:.2g} seconds\n".format(src_file, time_elapsed))
 
     return amount_of_code, time_elapsed
+
+
+def gen_binder_url(fname, binder_conf):
+    if not isinstance(binder_conf, dict):
+        return None
+    req_values = ['url', 'org', 'repo', 'branch', 'dependencies']
+
+    # Ensure all fields are populated
+    missing_values = [val for val in req_values
+                      if not isinstance(binder_conf.get(val, None), str)]
+    if len(missing_values) > 0:
+        raise ValueError('binder_conf is missing values for: {}'.format(
+            missing_values))
+
+    # Ensure we have a requirements file
+    allowed_reqs_files = ['requirements.txt', 'environment.yml']
+    path_reqs = binder_conf['dependencies']
+    if not any(path_reqs.endswith(ii) for ii in allowed_reqs_files):
+        raise ValueError('Requirements file must be one of {}'.format(
+            allowed_reqs_files))
+
+    if not any(binder_conf['url'].startswith(ii)
+               for ii in ['http://', 'https://']):
+        raise ValueError('did not supply a valid url, '
+                         'gave url: {}'.format(binder_conf['url']))
+
+    # Build the URL
+    binder_fpath = '_downloads/{}'.format(fname.replace('.py', '.ipynb'))
```

Comment:
ping - could you explain what you're suggesting here? Do you want to change the naming convention to the `ipynb` file itself so that it includes the full path?